source/plugin/commands/arguments.rst

```diff
@@ -25,13 +25,15 @@ it with :javadoc:`CommandContext#getOne(String)`. Optional and weak arguments ma
 
 Many of the parsers may return more than one object (e.g. multiple players with a matching username). In that case, you
 must use the :javadoc:`CommandContext#getAll(String)` method to get the ``Collection`` of possible matches.
-**Otherwise, the context object will throw an exception!**
+**Otherwise, the context object will throw an exception!** Please make sure that you make a conscious decision whether
+you need to limit your ``CommandExecutor`` to a single result or can support multiple arguments. If you support multiple
+arguments the users have to type less and shorter commands. Example: ``/tell @a Who took the cookies?``.
 
 .. tip::
 
-   You can use the
-   :javadoc:`GenericArguments#onlyOne(CommandElement)` element to limit the amount of returned values to a single one,
-   so you can safely use ``args.<T>getOne(String)``.
+   You can use the :javadoc:`GenericArguments#onlyOne(CommandElement)` element to limit the amount of returned values to
```

Comment:
It doesn't _limit_ the number returned, that implies that if two values were to be returned, one would be instead. It ensures only one value can be returned, throwing an exception if more than one value would be returned.

I realise that the docs say that now, but we're better off correcting this while we're here.